lvjit/LuaFunctionBuilder.cpp

```diff
@@ -1624,39 +1685,234 @@ bool Lua::FunctionBuilder::do_testset(TR::BytecodeBuilder* builder, TR::Bytecode
+   inlineluaD_precall->Store("diff",
+   inlineluaD_precall->      ConvertTo(Int32,
+   inlineluaD_precall->      Div(
+   inlineluaD_precall->          Sub(
+   inlineluaD_precall->              ConvertTo(Int64,
+   inlineluaD_precall->                        Load("stack_last")),
+   inlineluaD_precall->              ConvertTo(Int64,
+   inlineluaD_precall->                        Load("top"))),
+   inlineluaD_precall->      ConstInt64(luaTypes.TValue->getSize()))));
+
+   //fsize - 1 > diff is the same as diff <= fsize
+   TR::IlBuilder *growrequired = nullptr;
+   inlineluaD_precall->IfThen(&growrequired,
+   inlineluaD_precall->       GreaterThan(
+   inlineluaD_precall->                   Sub(
+   inlineluaD_precall->                       Load("fsize"),
+   inlineluaD_precall->                       ConstInt32(1)),
+   inlineluaD_precall->                   Load("diff")));
+
+   growrequired->Store("ra",
+   growrequired->      Call("jit_checkstackp", 3,
+   growrequired->           Load("L"),
+   growrequired->           Load("fsize"),
+   growrequired->           Load("ra")));
+
+   jit_Protect(growrequired);
+
+   //if (p->is_vararg != 1) {  /* do not use vararg? */
+   inlineluaD_precall->Store("is_vararg",
+   inlineluaD_precall->      ConvertTo(Int32,
+   inlineluaD_precall->                LoadIndirect("Proto", "is_vararg",
+   inlineluaD_precall->                             Load("p"))));
+
+   TR::IlBuilder *notvararg = nullptr;
+   TR::IlBuilder *vararg = nullptr;
+   inlineluaD_precall->IfThenElse(&notvararg, &vararg,
+   inlineluaD_precall->       NotEqualTo(
+   inlineluaD_precall->                  Load("is_vararg"),
+   inlineluaD_precall->                  ConstInt32(1)));
+
+   notvararg->Store("numparams",
+   notvararg->      ConvertTo(Int32,
+   notvararg->                LoadIndirect("Proto", "numparams",
+   notvararg->                             Load("p"))));
+
+   //for (; n < p->numparams; n++)
+   TR::IlBuilder *setnils = nullptr;
+   notvararg->ForLoopUp("n", &setnils,
+   notvararg->          Load("n"),
+   notvararg->          Load("numparams"),
+   notvararg->          ConstInt32(1));
 
-   // if (luaD_precall(L, ra, nresults))
-   auto isCFunc = builder->Call("luaD_precall", 3,
-                  builder->     Load("L"),
-                  builder->     Load("ra"),
-                  builder->     Load("nresults"));
+   //setnilvalue(L->top++);
+   setnils->StoreIndirect("TValue", "tt_",
+   setnils->             LoadIndirect("lua_State", "top",
+   setnils->                          Load("L")),
+   setnils->             ConstInt32(LUA_TNIL));
+
+   setnils->StoreIndirect("lua_State", "top",
+   setnils->             Load("L"),
+   setnils->             Add(
+   setnils->                ConvertTo(Int64,
+   setnils->                          LoadIndirect("lua_State", "top",
+   setnils->                                       Load("L"))),
+   setnils->                ConstInt64(luaTypes.TValue->getSize())));
```

Comment:
Yes that makes more sense.  Copied the code from somewhere else.  In a future change I will fix the location where I copied this code from